Lib/test/_test_multiprocessing.py

```diff
@@ -4706,12 +4706,216 @@ def is_alive(self):
             any(process.is_alive() for process in forked_processes))
 
 
-
 class MiscTestCase(unittest.TestCase):
     def test__all__(self):
         # Just make sure names in blacklist are excluded
         support.check__all__(self, multiprocessing, extra=multiprocessing.__all__,
                              blacklist=['SUBDEBUG', 'SUBWARNING'])
+
+
+class TestSyncManagerTypes(unittest.TestCase):
+    """Test all the types which can be shared between parent and child
+    process by using a manager which acts as an intermediary between
+    the two.
+
+    In the following tests the base type is created in the parent
+    process. "callback" function represents how the base type is
+    received (and handled) from within the child process. E.g.:
+
+    def test_list(self):
+        def worker(obj):
+            # === within the child process ===
+            assert obj[0] == 1
+
+        # === within the parent process ===
+        o = self.manager.list()
+        o.append(1)
+        self.run_test(worker, o)
+    """
+    manager_class = multiprocessing.managers.SyncManager
+
+    def setUp(self):
+        self.manager = self.manager_class()
+        self.manager.start()
+        self.proc = None
+
+    def tearDown(self):
+        self.manager.shutdown()
+        if self.proc is not None:
+            self.proc.terminate()
+            self.proc.join()
+
+    def run_test(self, worker, obj):
+        self.proc = multiprocessing.Process(target=worker, args=(obj, ))
+        self.proc.start()
+        self.proc.join()
```

Comment:
Should this raise if the process' exit code is non-zero?

Reply:
Yup, will add `self.assertEqual(self.proc.exitcode, 0)` so that test will fail (as opposed to just print a traceback).